provider/netperf_test.py

```diff
@@ -57,25 +57,25 @@ def netperf_stress(test, params, vm):
                           params.get_numeric("netperf_para_sessions"),
                           params.get("netperf_cmd_prefix", ""),
                           package_sizes=params.get("netperf_sizes"))
-        if utils_misc.wait_for(n_client.is_netperf_running, 10, 0, 1,
+        if utils_misc.wait_for(n_client.is_netperf_running, 10, 0, 0.1,
                                "Wait netperf test start"):
             logging.info("Netperf test start successfully.")
         else:
             test.error("Can not start netperf client.")
         start_time = time.time()
-        execution_time = test_duration + deviation_time
-        utils_misc.wait_for(lambda: not
-                            n_client.is_netperf_running(),
-                            execution_time, 0, 2,
-                            "Wait netperf test finish %ss" % test_duration)
-        stop_time = time.time()
-        run_time = stop_time - start_time
+        duration = time.time() - start_time
+        while duration < test_duration + deviation_time:
+            status = n_client.is_netperf_running()
+            if not status and duration < test_duration - deviation_time:
```

Comment:
Same here, it is not necessary to calculate it every time in the loop.

Reply:
I have changed it. Please help me review it again. Thanks  a lot.